src/AlgebraicGeometry/Surfaces/EnriquesAut.jl

```diff
@@ -0,0 +1,908 @@
+  Gminus, inc_Gminus = kernel(inc)
+  @vprintln :EnriquesAuto 2 "done"
+  ECtx.orderGbar = order(Gminus)
+
+  # compute Gplus
+  Q = orthogonal_submodule(L26,SY)
+  DQ = discriminant_group(Q)
+  glue_SY_Q,inc1,inc2 = glue_map(L26,SY, Q)
+  DO = codomain(glue_SY_Q)
+
+  @vprint :EnriquesAuto 2 "preparing membership test\n"
+  DSY = discriminant_group(SY)
+  ODSY = orthogonal_group(DSY)
+  tmp = [preimage(phiSm, i) for i in small_generating_set(Gminus)]
+  tmp_hom = [hom(DO,DO, [DO(lift(j)*i) for j in gens(DO)]) for i in tmp]
+  gens_Gplus = elem_type(ODSY)[ODSY(inv(inc1)*glue_SY_Q*i*inv(glue_SY_Q)*inc1) for i in tmp_hom]
+  Gplus,_ = sub(ODSY, gens_Gplus)
+  ECtx.Gplus = Gplus
+  @assert order(Gplus) == order(Gminus)
+  # iso_Gplus_minus = hom(Gplus, Gminus, gens(Gplus), small_generating_set(Gminus))
+
+  Dplus = codomain(phi)
+  ODplus = orthogonal_group(Dplus)
+  # turn this into a mod 2 matrix group
+  # with respect to the basis of SY mod 2
+  B = 1//2*basis_matrix(SY)
+  gens_Gplus_mat = FqMatrix[]
+  for g in gens(Gplus)
+    g2 = reduce(vcat, [change_base_ring(GF(2), solve(B, matrix(QQ, 1, 26, lift(g(DSY(vec(B[i, :]))))); side=:left))  for i in 1:10])
+    push!(gens_Gplus_mat, g2)
+  end
+  Gplus_mat = matrix_group(GF(2), 10, gens_Gplus_mat)
+  ECtx.Gplus_mat = Gplus_mat
+
+  snf_Dplus, i_snf = snf(Dplus)
+  gens_Dplus = [change_base_ring(GF(2), solve(basis_matrix(SY), matrix(QQ, 1, 26, 2*lift(inc_Dplus(i_snf(i)))); side=:left)) for i in gens(snf_Dplus)]
+
+  gens_Dplus = reduce(vcat, gens_Dplus)
+  ECtx.Dplus = gens_Dplus
+
+
+  Dplus_perp, _ = orthogonal_submodule(DSY, Dplus)
+  Dplus_perp,_ = snf(Dplus_perp)
+
+  # needed for hash
+  ECtx.Dplus_perp = reduce(vcat,[change_base_ring(GF(2), 2*solve(basis_matrix(SY), matrix(QQ, 1, 26, lift(i)); side=:left)) for i in gens(Dplus_perp)])
+
+
+  @vprint :EnriquesAuto 2 "done\n"
+  ECtx.membership_test = membership_test_as_group
+  #upper_bound = mass(ECtx)*length(initial_automorphisms(ECtx))
+  return ECtx
+end
+  
+@doc raw"""
+    EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat) -> EnriquesBorcherdsCtx 
+  
+Return a context object for Borcherds' method for Enriques surfaces. 
+  
+Let ``\pi: X \to Y`` be the universal cover of an Enriques surface and ``\epsilon`` the Enriques involution. 
+# Input: 
+- `SY2` -- the invariant lattice of the Enriques involution in the numerical lattice `SX` of ``X``. 
+"""
+function EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat; ample=nothing)
+  @req is_sublattice(SY2, SX) "SY2 must be a sublattice of SX"
+  tmp = rescale(SY2, 1//2)
+  @req is_unimodular(tmp) && is_even(tmp) && signature_tuple(tmp) == (1, 0, 9) "SY2 must be isomorphic to E10(2)"
+  B = solve(basis_matrix(SX), basis_matrix(SY2); side=:left)
+  L26, SX1, iSX1 = embed_in_unimodular(lattice(rational_span(SX)), 1, 25, primitive=true,even=true)
+  SY1 = lattice(ambient_space(SX1),solve(basis_matrix(SX),basis_matrix(SY2);side=:left)*basis_matrix(SX1))
+  if ample==nothing 
```

Comment:
```suggestion
  if ample === nothing 
```
`nothing` should be compared with `===` or `isnothing` for better performance